src/parser/ast/ObjectExpressionNode.h

```diff
@@ -41,9 +41,104 @@ class ObjectExpressionNode : public ExpressionNode {
+                    context->giveUpRegister(); // key
+                } else {
+                    // handle spread element in Object initialization
+                    ASSERT(property->astNode()->type() == ASTNodeType::SpreadElement);
+                    Node* element = property->astNode()->asSpreadElement()->argument();
+
+                    ByteCodeRegisterIndex elementIndex = element->getRegister(codeBlock, context);
+                    ByteCodeRegisterIndex dataIndex = context->getRegister();
+                    ByteCodeRegisterIndex keyIndex = context->getRegister();
+                    ByteCodeRegisterIndex valueIndex = context->getRegister();
+
+                    element->generateExpressionByteCode(codeBlock, context, elementIndex);
+
+                    codeBlock->pushCode<JumpIfUndefinedOrNull>(JumpIfUndefinedOrNull(ByteCodeLOC(m_loc.index), false, elementIndex), context, this->m_loc.index);
+                    size_t pos = codeBlock->lastCodePosition<JumpIfUndefinedOrNull>();
+
+                    codeBlock->pushCode(CreateEnumerateObject(ByteCodeLOC(m_loc.index), elementIndex, dataIndex, true), context, this->m_loc.index);
+
+                    size_t checkPos = codeBlock->currentCodeSize();
+                    codeBlock->pushCode(CheckLastEnumerateKey(ByteCodeLOC(m_loc.index)), context, this->m_loc.index);
+                    codeBlock->peekCode<CheckLastEnumerateKey>(checkPos)->m_registerIndex = dataIndex;
+                    codeBlock->pushCode(GetEnumerateKey(ByteCodeLOC(m_loc.index), keyIndex, dataIndex), context, this->m_loc.index);
+
+                    codeBlock->pushCode(GetObject(ByteCodeLOC(m_loc.index), elementIndex, keyIndex, valueIndex), context, this->m_loc.index);
+                    codeBlock->pushCode(CreateObjectPrepare(ByteCodeLOC(m_loc.index), false, objectCreationDataIndex, keyIndex, valueIndex, false), context, this->m_loc.index);
+                    codeBlock->pushCode(Jump(ByteCodeLOC(m_loc.index), checkPos), context, this->m_loc.index);
+
+                    size_t exitPos = codeBlock->currentCodeSize();
+                    codeBlock->peekCode<CheckLastEnumerateKey>(checkPos)->m_exitPosition = exitPos;
+
+                    // for drop elementIndex, dataIndex, keyIndex, valueIndex
+                    context->giveUpRegister();
+                    context->giveUpRegister();
+                    context->giveUpRegister();
+                    context->giveUpRegister();
+
+                    codeBlock->peekCode<JumpIfUndefinedOrNull>(pos)->m_jumpPosition = codeBlock->currentCodeSize();
+                }
+            }
+
+            codeBlock->pushCode(CreateObject(ByteCodeLOC(m_loc.index), dstRegister, objectCreationDataIndex), context, this->m_loc.index);
+            if (m_properties.size()) {
+                codeBlock->peekCode<CreateObjectPrepare>(initCodePosition)->m_propertyReserveSize = propertyIndex;
+                context->giveUpRegister();
+            }
```

Comment:
This if statement could be removed too

Reply:
Oops! this if statement was needed for testing.
I am going to remove it